consensus/src/round_manager.rs

```diff
@@ -933,8 +966,21 @@ impl RoundManager {
 
         observe_block(proposal.timestamp_usecs(), BlockStage::SYNCED);
 
+        // Since processing proposal is delayed due to backpressure or payload availability, we add
+        // the block to the block store so that we don't need to fetch it from remote once we
+        // are out of the backpressure. Please note that delayed processing of proposal is not
+        // guaranteed to add the block to the block store if we don't get out of the backpressure
+        // before the timeout, so this is needed to ensure that the proposed block is added to
+        // the block store irrespective. Also, it is possible that delayed processing of proposal
+        // tries to add the same block again, which is okay as `execute_and_insert_block` call
+        // is idempotent.
+        self.block_store
+            .insert_block(proposal.clone())
```

Comment:
it also doesn't look like we dedup block from the same round

Reply:
It's here: https://github.com/aptos-labs/aptos-core/blob/ffb925d00e5d00741d5f8d9b76828022d2247a26/consensus/src/round_manager.rs#L935-L940